Implement Bug Bounty page

wackerow opened the pull request:
## Description
- Implements `/bug-bounty` and affiliated components

## Todo
- Fix i18n strings for path breadcrumbs
- Fix image width—should be 24px wide
- Check `<code>` styling compared to Gatsby repo

## Related Issue
https://www.notion.so/efdn/Migrate-React-pages-8acee0dc80d04e55b109e9852c20ef2f?pvs=4

## Preview link
https://deploy-preview-144--ethereum-org-fork.netlify.app/en/bug-bounty

wackerow committed:
update Leaderboard component

wackerow committed:
initalize Bug Bounty page

wackerow committed:
convert bounty hunter data to JSON

wackerow committed:
update Breadcrumbs component and story

wackerow committed:
update BugBountyCards component

wackerow committed:
update bug-bounty related components and strings

wackerow committed:
update /bug-bounty page component

wackerow committed:
patch crumbs type

todo: fix i18n strings for path breadcrumbs

wackerow committed:
patch i18n string

wackerow committed:
patch image sizes

wackerow committed:
Update CODEOWNERS

point to new JSON data file

fredrik0x reviewed (approved):
LGTM. minor nitpick would be that "Bug Bounty Program 🐛" is cut off a bit short at the bottom (the "g" and "y" is a bit short)

minimalsm commented on src/pages/bug-bounty.tsx:
```diff
@@ -0,0 +1,896 @@
+
+  const clientsSmall = clients.map((client) => {
+    const thumbnail = client.image as StaticImageData
+    thumbnail.width = 24
```
I'm seeing a minimum width of 60px in prod

Prod:
<img width="945" alt="Screenshot 2023-11-28 at 12 57 07" src="https://github.com/ethereum/ethereum-org-fork/assets/62268199/86d41211-277c-4ee2-ab2d-01f8a178fe1a">

This PR:
<img width="810" alt="Screenshot 2023-11-28 at 12 57 11" src="https://github.com/ethereum/ethereum-org-fork/assets/62268199/2c7c2682-e630-47dd-ba5e-4c010758c988">

wackerow replied:
Patched!

pettinarip commented on src/components/Breadcrumbs/index.tsx:
```diff
@@ -52,33 +52,34 @@ const Breadcrumbs: React.FC<IProps> = ({
       return {
         fullPath: slugChunk.slice(0, idx + 2).join("/") + "/",
-        text: t(path),
+        text: t(path), // TODO: fix i18n strings for path breadcrumbs
```
I think most of the translations for breadcrumbs are in the `common` ns. Maybe we could import the `common` ns and use it here
```ts
const { t } = useTranslation(["common", "page-index"])

...
text: t(path, { ns: "common" })
```

Just an idea, we can tackle this in a separate PR.

wackerow replied:
Okay... so turns out we do not have a string for `"bug-bounty": "Bug bounty",` in our `common.json` file yet... This is also a bug in production, but easy fix.. I'll add the string in English and this should get resolved in time with future Crowdin translations.

By adding this it works locally without adding "common" to the namespaces here

wackerow replied:
Ooops, I take that back... need to add "common", but only to the `useTranslation` argument array. Will leave it in

pettinarip commented on src/pages/bug-bounty.tsx:
```diff
@@ -0,0 +1,896 @@
+  const { locale } = context
+
+  // load i18n required namespaces for the given page
+  const requiredNamespaces = getRequiredNamespacesForPath("bug-bounty")
+
+  return {
+    props: await serverSideTranslations(locale!, requiredNamespaces),
+  }
+}
```
We need to pass the `lastDeployDate` prop to display the date correctly.

```suggestion
  const requiredNamespaces = getRequiredNamespacesForPath("bug-bounty")
  const lastDeployDate = getLastDeployDate()

  return {
    props: {
      ...(await serverSideTranslations(locale!, requiredNamespaces)),
      lastDeployDate
    },
  }
}
```

pettinarip commented on src/pages/bug-bounty.tsx:
```diff
@@ -0,0 +1,896 @@
+  return b.score - a.score
+}
+
+export const getStaticProps: GetStaticProps<{}, {}> = async (context) => {
```
Lets use the `satisfies` form, like in [the slug file](https://github.com/ethereum/ethereum-org-fork/blob/dev/src/pages/%5B...slug%5D.tsx#L96).
```suggestion
export const getStaticProps = async (context) => {
```

pettinarip commented on src/lib/constants.ts:
```diff
@@ -25,6 +25,7 @@ export const CONTENT_IMAGES_MAX_WIDTH = 800
 export const GITHUB_BASE_API =
   "https://api.github.com/repos/ethereum/ethereum-org-website"
 export const GITHUB_COMMITS_URL = GITHUB_BASE_API + "/commits"
+export const GITHUB_ROOT = `https://github.com/`
```
nitpick
```suggestion
export const GITHUB_URL = `https://github.com/`
```

pettinarip commented on src/components/Leaderboard.tsx:
```diff
@@ -54,14 +53,12 @@ const Leaderboard: React.FC<IProps> = ({ content, limit = 100 }) => {
+        .map(({ name, username, score }, idx) => {
+          const hasGitHub = !!username
+          const avatarImg =
+            GITHUB_ROOT + (username || "random") + ".png?size=40"
```
💯

wackerow committed:
add lastDeployDate to getStaticProps

wackerow committed:
Merge branch 'dev' into bug-bounty-page

wackerow committed:
Create resizeImage.ts

wackerow committed:
patch image sizes; linting

wackerow committed:
style syntax adjustments

adds overflow=auto to fix text cutting off at baseline

wackerow commented:
Thanks @fredriksvantes! This was a bug in production already, but I added a small style adjustment to fix this, thanks for surfacing!

wackerow committed:
apply suggestions from code review

fix missing i18n string for Bug Bounty breadcrumb text

wackerow committed:
rm use of Translation component

wackerow committed:
eslint-disable react/no-unescaped-entities

temporary, revert if #153 is merged, or remove and fix escaped characters if not

wackerow committed:
Revert "eslint-disable react/no-unescaped-entities"

This reverts commit 444cc907be24061fc58965af4ac16b1672c2d127.

wackerow committed:
escape characters

pettinarip reviewed (approved):
🎉
